### ISSUE TITLE
Reviews by Product: list items have incorrect aria-label

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#11974]
Blocks https://github.com/woocommerce/woocommerce-blocks/pull/11975.

## To reproduce

1. Create a post or page.
2. Add the Reviews by Product block.
3. Using the browser Inspector (<kbd>F12</kbd>), select one of the checkboxes.
4. Notice the value of the `aria-label` shows placeholder signs: `%1$s` and `%2$2`.
5. There is also an error in the console: `TypeError: [sprintf] expecting number but found undefined`.

![imatge](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/52f6ad74-f0f2-41a3-8a7e-9e9c07434375)

## Expected behavior

There should be no error and the `aria-label` value should show the product name and the number of reviews.
